### PR TITLE
Use pre-commit to run code checks (also in CI)

### PR DESCRIPTION
**Pull request by erlendvollset**
This PR enables running code checks as pre-commit hooks for a shorter feedback loop. Install the hooks by running:
```
pre-commit install
```

We also now run all the code checks as part of the CI pipeline. 

I took the liberty of adding isort and autoflake aswell - these will sort and remove unused imports.

### CODE REVIEW

File: tests/test_map.py [hunk 1]
```diff
@@ -75,14 +76,6 @@ def test_map_remove(xs: Dict[str, int]):
     assert len(m) == count == 0
 
 
-@given(st.dictionaries(keys=st.text(), values=st.integers()))
-def test_map_to_seq(xs: Dict[str, int]):
```
Comment (erlendvollset) [inline tests/test_map.py:79]: duplicate

File: .flake8 [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [flake8]
 ignore = E731, T484, T400 # Do not assign a lambda expression, use a def
-max-line-length = 120
\ No newline at end of file
+max-line-length = 121
```
Comment (erlendvollset) [inline .flake8:3]: needs to be 121 or it disagrees with black.

File: tests/test_map.py (hunk 1 above)
Comment (dbrattli) [reply tests/test_map.py:79]: I have fixed this locally. Will push.

File: tests/test_map.py [hunk 3]
```diff
@@ -46,7 +47,7 @@ def test_map_to_seq(xs: Dict[str, int]):
     items: ItemsView[str, int] = xs.items()
     ys = map.of_seq(items)
     zs = pipe(ys, map.to_seq)
-    assert list(xs) == list(zs)
```
Comment (erlendvollset) [inline tests/test_map.py:49]: this was not silently failing due to the duplicate test below

Comment (dbrattli) [reply tests/test_map.py]: Fixed

**Review by dbrattli — APPROVED** [review]
This looks great!

File: .flake8 (hunk 2 above)
Comment (dbrattli) [reply .flake8:3]: Looks good!

**Pull request merged**